server/core/storage.go

```diff
@@ -159,6 +160,19 @@ func (s *Storage) LoadRegions(f func(region *RegionInfo) []*RegionInfo) error {
 	return loadRegions(s.Base, f)
 }
 
+// LoadRegionsOnce loads all regions from storage to RegionsInfo.Only load one time from regionStorage.
+func (s *Storage) LoadRegionsOnce(f func(region *RegionInfo) []*RegionInfo) error {
+	if atomic.LoadInt32(&s.useRegionStorage) == 0 {
+		return loadRegions(s.Base, f)
+	}
+	if atomic.CompareAndSwapInt32(&s.regionLoaded, 0, 1) {
```

Comment:
I'm sorry about letting u use `atomic`.  actually, there is a concurrency problem,  `loadRegions` may be failed or spent a long time, other goroutines think its' already loaded. PTAL @mantuliu

Reply:
Yes, this is a concurrency issue. I also ignored this before, and I changed it back to lock later. Thank you very much for your rigor.

Reply:
PTAL @nolouch.